raft/log_test.go

```diff
@@ -250,137 +252,150 @@ func TestState_String(t *testing.T) {
 
 // Cluster represents a collection of nodes that share the same mock clock.
 type Cluster struct {
-	Nodes []*Node
+	Logs []*Log
 }
 
-// NewCluster creates a new cluster with an initial set of nodes.
-func NewCluster(nodeN int) *Cluster {
+// NewCluster creates a new 3 log cluster.
+func NewCluster() *Cluster {
 	c := &Cluster{}
-	for i := 0; i < nodeN; i++ {
-		n := c.NewNode()
-		n.Open()
-
-		// Initialize the first node.
-		// Join remaining nodes to the first node.
-		if i == 0 {
-			go func() { n.Clock().Add(2 * n.Log.ApplyInterval) }()
-			if err := n.Log.Initialize(); err != nil {
-				panic("initialize: " + err.Error())
-			}
-		} else {
-			go func() { n.Clock().Add(n.Log.HeartbeatInterval) }()
-			if err := n.Log.Join(c.Nodes[0].Log.URL); err != nil {
-				panic("join: " + err.Error())
-			}
-		}
-	}
-
-	// Make sure everything is replicated to all followers.
-	c.Nodes[0].Log.Flush()
-	c.Clock().Add(c.Nodes[0].Log.HeartbeatInterval)
+	t := NewTransport()
+
+	logN := 3
+	for i := 0; i < logN; i++ {
+		l := NewLog(&url.URL{Host: fmt.Sprintf("log%d", i)})
+		l.Transport = t
+		c.Logs = append(c.Logs, l)
+		t.register(l.Log)
+		warnf("Log %s: %p", l.URL.String(), l.Log)
+	}
+	warn("")
```

Comment:
Yes, it's nice to see the logs and their pointer in one section at the top. There's so much logging that happens with trace that it makes it easier to see.